soh/src/overlays/actors/ovl_En_Zo/z_en_zo.c

```diff
@@ -427,7 +427,8 @@ u16 func_80B61024(PlayState* play, Actor* thisx) {
                 return 0x402D;
             }
 
-            if (Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED)) {
+            if (!CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED) ||
+                CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_OBTAINED_RUTOS_LETTER)) {
```

Comment:
after reading this through, i don't think we need to check for `!gFixZoraHintDialogue` here. i'm pretty sure it can be simplified to

```suggestion
            if (Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED) || (CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_OBTAINED_RUTOS_LETTER))) {
```

Reply:
I think the idea was that we dont want this text to appear after king zora has moved with the fix, instead we want it to appear once obtaining the letter. That being said, I guess the current implementation still doesnt fix the issue of this text appearing once king zora has moved but before obtaining the sapphire. that would then be

```
if (!CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED) ||
     CVarGetInteger("gFixZoraHintDialogue", 0) && Flags_GetEventChkInf(EVENTCHKINF_OBTAINED_RUTOS_LETTER) && !Flags_GetEventChkInf(EVENTCHKINF_KING_ZORA_MOVED)) {
```

Though maybe I'm overthinking how much of a 'fix' this needs to be now..

Reply:
I disagree with "this text _**should not**_ display after King Zora moves". It is more that the text just needs to be displayed "sooner". If the text isn't displayed, then the previous text ID is used, and considering the progression of the game, I don't think it makes sense to go back to older dialog, that doesn't really happen anywhere else in the game.

Bria is just suggesting the above because the text would display regardless once KZ moves, so the CVar check can just be short-circuited out.

Reply:
if thats the case then yea I agree it can be removed 👍🏼

Reply:
To hammer it home, if the text were not displayed after King Zora moved, it would go back to telling you about searching Lake Hylia, which doesn't make sense as a hint and would point players in the wrong direction. Keeping the hint here still pushes players toward King Zora and thus toward Zora's Fountain in case they've forgotten that he mweep'd it open for them.